config/validate/validate.go

```diff
@@ -223,31 +235,33 @@ func validateStruct(vObj reflect.Value, ast astnode.AstNode, source []byte, chec
 				dupLists[dupListKey] = map[string]struct{}{}
 			}
 
-			for i := 0; i < f.Value.Len(); i++ {
-				key := ""
-				if f.Value.Index(i).Kind() == reflect.String {
-					key = f.Value.Index(i).Convert(reflect.TypeOf("")).Interface().(string)
-				} else {
-					key = f.Value.Index(i).Interface().(keyed).Key()
-				}
-				if _, alreadyDefined := dupLists[dupListKey][key]; alreadyDefined {
-					// duplicate entry!
-					line, col, highlight := 0, 0, ""
-					if sub_node != nil {
-						sub_sub_node, ok := sub_node.SliceChild(i)
-						if sub_sub_node != nil && ok {
-							line, col, highlight = sub_sub_node.ValueLineCol(src)
+			if _, ignored := ignoreDupsList[f.Type.Name]; !ignored {
```

Comment:
minor: can we reuse `dupListKey` rather than `f.Type.Name`?

Reply:
That would work yes, but I'm not sure it's clearer. Keeping it as `f.Type.Name` indicates that the ignorelists are not impacted by the merged keys lists (granted, they should never overlap).